### PR TITLE
Simpler dependencies

### PR DESCRIPTION
**Pull request by dvdplm**
Micro PR with outcome of some code reading and tinkering.

Moves `sp-arithmetic` to `dev-dependencies` and switches to use `default-features = false`. Was hoping for a reduced number of dependent crates but it stays the same.﻿

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,14 +34,14 @@ url = "2.2.1"
...
-sp-core = { package = "sp-core", git = "https://github.com/paritytech/substrate/", branch = "master" }
-sp-runtime = { package = "sp-runtime", git = "https://github.com/paritytech/substrate/", branch = "master" }
+sp-core = { package = "sp-core", git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false  }
+sp-runtime = { package = "sp-runtime", git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false }
 sp-version = { package = "sp-version", git = "https://github.com/paritytech/substrate/", branch = "master" }
 
 frame-metadata = "14.0.0"
 
 [dev-dependencies]
+sp-arithmetic = { package = "sp-arithmetic", git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false }
```
Comment (niklasad1) [inline Cargo.toml:44]:
```suggestion
sp-arithmetic = { git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false }
```

File: Cargo.toml [hunk 2]
```diff
@@ -34,14 +34,14 @@ url = "2.2.1"
 
 subxt-macro = { version = "0.1.0", path = "macro" }
 
-sp-arithmetic = { package = "sp-arithmetic", git = "https://github.com/paritytech/substrate/", branch = "master" }
-sp-core = { package = "sp-core", git = "https://github.com/paritytech/substrate/", branch = "master" }
-sp-runtime = { package = "sp-runtime", git = "https://github.com/paritytech/substrate/", branch = "master" }
+sp-core = { package = "sp-core", git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false  }
```
Comment (niklasad1) [inline Cargo.toml:37]:
```suggestion
sp-core = { git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false  }
```

File: Cargo.toml [hunk 3]
```diff
@@ -34,14 +34,14 @@ url = "2.2.1"
...
+sp-runtime = { package = "sp-runtime", git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false }
```
Comment (niklasad1) [inline Cargo.toml:38]:
```suggestion
sp-runtime = { git = "https://github.com/paritytech/substrate/", branch = "master", default-features = false }
```

**Review by jsdw — APPROVED** [review]
LGTM! (I like Niklas's suggestions also)

**Review by insipx — APPROVED** [review]

**Pull request merged**